### PR TITLE
Netifyd

### PR DESCRIPTION
**Pull request by gsanchietti**
- [x] import netifyd 4.2.0
- [x] compile netify-flow-actions plugin
- [x] netify-plugin-stats plugin

### CODE REVIEW

File: docs/build.md [hunk 1]
```diff
@@ -174,3 +174,91 @@ Publish the image:
 buildah login ghcr.io
 buildah push ghcr.io/nethserver/nextsecurity-builder docker://ghcr.io/nethserver/nextsecurity-builder
 ```
+
+## Netifyd plugins
+
+NextSecurity uses two [netifyd](https://gitlab.com/netify.ai/public/netify-agent) proprietaries plugins from [Netify](https://www.netify.ai/):
```
Comment (filippocarletti) [inline docs/build.md:180]:
```suggestion
NextSecurity uses two [netifyd](https://gitlab.com/netify.ai/public/netify-agent) proprietary plugins from [Netify](https://www.netify.ai/):
```

File: docs/build.md [hunk 2]
```diff
@@ -174,3 +174,91 @@ Publish the image:
...
+
+- Netify Flow Actions Plugin (netify-flow-actions)
+- Netify Agent Stats Plugin (netify-plugin-stats)
+
+The plugins should be used with the latest netifyd stable version (4.2.2 at time of writing).
```
Comment (filippocarletti) [inline docs/build.md:185]:
```suggestion
The plugins should be used with the latest netifyd stable version (4.2.2 at the time of writing).
```

File: docs/build.md [hunk 3]
```diff
@@ -174,3 +174,91 @@ Publish the image:
...
+To create the files for the build, follow the below steps. Such steps should be needed only after a netifyd/plugin version change.
```
Comment (filippocarletti) [inline docs/build.md:186]:
```suggestion
To create the files for the build, follow the steps below. Such steps should be needed only after a netifyd/plugin version change.
```

File: files/etc/uci-defaults/99-nextsec-netifyd [hunk 4]
```diff
@@ -0,0 +1,7 @@
+if [ -f /usr/lib/libnetify-flow-actions.so.0.0.0 ] && ! grep -q np-nfa /etc/netifyd.conf; then
```
Comment (filippocarletti) [inline files/etc/uci-defaults/99-nextsec-netifyd:1]:
```suggestion
if [ -f /usr/lib/libnetify-flow-actions.so.0 ] && ! grep -q np-nfa /etc/netifyd.conf; then
```

File: files/etc/uci-defaults/99-nextsec-netifyd [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+    (
+        echo
+        echo "[plugin_detections]"
+        echo "np-nfa = /usr/lib/libnetify-flow-actions.so.0.0.0"
```
Comment (filippocarletti) [inline files/etc/uci-defaults/99-nextsec-netifyd:5]:
```suggestion
        echo "np-nfa = /usr/lib/libnetify-flow-actions.so.0"
```

File: packages/netify-plugin-stats/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# netify-plugin-stats
+
+The stats plugins is disabled by default.
+To enable it, execute:
+```bash
+if [ -f /usr/lib/libnetify-plugin-stats.so.0.0.0 ] && ! grep -q np-stats /etc/netifyd.conf; then
```
Comment (filippocarletti) [inline packages/netify-plugin-stats/README.md:6]:
```suggestion
if [ -f /usr/lib/libnetify-plugin-stats.so.0 ] && ! grep -q np-stats /etc/netifyd.conf; then
```

File: packages/netify-plugin-stats/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    (
+        echo
+        echo "[plugin_stats]"
+        echo "np-stats = /usr/lib/libnetify-plugin-stats.so.0.0.0"
```
Comment (filippocarletti) [inline packages/netify-plugin-stats/README.md:10]:
```suggestion
        echo "np-stats = /usr/lib/libnetify-plugin-stats.so.0"
```

**Review by filippocarletti — CHANGES_REQUESTED** [review]

File: files/etc/uci-defaults/99-nextsec-netifyd (hunk 4 above)
Comment (gsanchietti) [reply files/etc/uci-defaults/99-nextsec-netifyd:1]: I do not get this one.
`/usr/lib/libnetify-flow-actions.so.0` is a link to `/usr/lib/libnetify-flow-actions.so.0.0.0`, why testing the presence of the link and then adding the link target to the configuration file?
I think it's not coherent.

File: files/etc/uci-defaults/99-nextsec-netifyd (hunk 5 above)
Comment (gsanchietti) [reply files/etc/uci-defaults/99-nextsec-netifyd:5]: See comment above

File: files/etc/uci-defaults/99-nextsec-netifyd (hunk 4 above)
Comment (gsanchietti) [reply files/etc/uci-defaults/99-nextsec-netifyd:1]: I do not get this one.
`/usr/lib/libnetify-flow-actions.so.0` is a link to `/usr/lib/libnetify-flow-actions.so.0.0.0`, why testing the presence of the link and then adding the link target to the configuration file?
I think it's not coherent.

File: packages/netify-plugin-stats/README.md (hunk 6 above)
Comment (gsanchietti) [reply packages/netify-plugin-stats/README.md:6]: See comment above

File: packages/netify-plugin-stats/README.md (hunk 7 above)
Comment (gsanchietti) [reply packages/netify-plugin-stats/README.md:10]: See comment above

File: files/etc/uci-defaults/99-nextsec-netifyd (hunk 4 above)
Comment (gsanchietti) [reply files/etc/uci-defaults/99-nextsec-netifyd:1]: Using the link should avoid configuration changes on minor releases.

File: files/etc/uci-defaults/99-nextsec-netifyd (hunk 5 above)
Comment (gsanchietti) [reply files/etc/uci-defaults/99-nextsec-netifyd:5]: @filippocarletti already tested it with the link.

**Pull request merged**